### DISCUSSION TITLE
Unable to ship Windows endpoint logs from an ec2 instance to ingest into Security Onion as an HIDS Experiment

### DISCUSSION THREAD

**Discussion by bouncebackbrownie** [Security-Onion-Solutions/securityonion#12785] [2.4]
### Version

2.4.60

### Installation Method

Other (please provide detail below)

### Description

installation

### Installation Type

Standalone

### Location

cloud

### Hardware Specs

other (please provide detail below)

### CPU

8

### RAM

32

### Storage for /

256

### Storage for /nsm

0

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hello everyone, I was making my own Cloud SOC lab as a project on AWS. As a part of that project, I wish to digest logs from a Windows ec2 instance endpoint to be able to monitor the endpoint activity as part of the HIDS component.

Collecting Endpoint Logs with Elastic Agent: https://www.youtube.com/watch?v=cGmQMsFuAvw&t=447s

I saw this video uploaded recently from Security Onion LLC and followed it accordingly as per my knoweldge. However, I was unable to see the endpoint logs being ingested and shown on my Security Onion Dashboard. I had already added  the CIDR block that I have established my Windows ec2 instance on as part of the VPC under 'firewall.hostgroups.elastic_agent_endpoint'. And even clicked on Options->Synchronize Grid at the top to push the changes into production.

However, after downloading my elastic agent, and right-clicking it to 'Run as Administrator', I saw errors like (on the SO-Elastic-Agent_Installer.txt file):-
timestamp=2024-04-11T17:21:13.2565196Z level=warn message="Installation Progress" FleetHostConnectivityCheck=Failed FleetHostURL=https://projectso-24:8220
timestamp=2024-04-11T17:21:13.2576566Z level=info message="Installation Progress" Status="No Fleet Hosts are accessible - Check connectivity to Fleet Host."
timestamp=2024-04-11T17:21:13.2577036Z level=info message="Installation Progress" Status="Exiting Installer..."

I am unable to figure out why it. 

I will attach some screenshots to help others guide me better.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/7edb166c-b72b-4bdf-8b1a-d88984690858)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/9d37a7ac-7f4b-4f48-850b-8912556005fb)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/050b49a0-55bc-4b71-a1b7-85e63efc3eb9)

[I had already updated the 'hosts' file under C:\Windows\System32\drivers\etc to add the IP and hostname for resolving purposes. The hostname is 'projectso-24']

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Is the agent able to connect to this `https://projectso-24:8220`?

**Reply by bouncebackbrownie** [reply]
It says: "timestamp=2024-04-11T17:21:13.2565196Z level=warn message="Installation Progress" FleetHostConnectivityCheck=Failed FleetHostURL=https://projectso-24:8220" (Please view the screenshot of SO-Elastic-Agent_Installer.txt file here)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/7dd1fae3-1513-4bff-a058-754f14911f7a)

The agent is exiting installation due to failure of 'FleetHostConnectivityCheck'. I request elaboration and guidance on this. As far as I know, we need to configure to add hosts on a Fleet server to receive and ingest endpoint logs right?

Checking when Grid Status is 'OK'
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/7db1eeea-95cc-4552-accc-ab5a0ad41e6b)

I am unable to reach through port 8220 on the hostname 'projectso-24' via HTTPS on browser (HTTP due to lack of SSL certificate)

 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/130464099/846005db-a211-4fdb-af67-01c6eccf4cd1)

**Reply by cm-ops** [reply]
Did you allow the firewall entry in SOC? https://docs.securityonion.net/en/2.4/elastic-agent.html

**Comment by jnhutchi3** [comment]
I encountered the same error on some of my endpoints -- "No Fleet Hosts are accessible - Check connectivity to Fleet Host."

However, I was able to successfully run the Elastic agent installation on some of my endpoints.  I'm not sure why the installation is working on some endpoints, but not others.  I'm running the same Windows operating system on all the endpoints and the firewall configuration should be the same for all the endpoints.  I'm able to connect to the Fleet Server from a web browser on all of the endpoints.
